packages/backend/src/QueryEngine.ts

```diff
@@ -186,4 +284,19 @@ export class QueryEngine<DB> {
             });
         }
     }
+
+    // TODO: fix the callback return type from AnyQuery to TQuery
+    // Not sure how to do this yet, or if this is even possible
+    registerQueries(queryFns: Array<(...params: any[]) => AnyQuery>) {
+        for (const queryFn of queryFns) {
+            const query = queryFn();
+
+            // TODO: add appropriate error method
+            if (!query.hash) {
+                throw new Error('Query to be registered is missing a hash!');
+            }
+
+            this.queries.set(query.hash, queryFn);
```

Comment:
throw an error if overriding: it's most likely an error. In fact two queries shouldn't have the same key.